File: base/src/main/java/com/zhangteng/base/mvp/base/BaseLoadingPresenter.java
package com.zhangteng.base.mvp.base;

public class BaseLoadingPresenter<V extends BaseLoadingView> extends BasePresenter<V> {
    V getBaseLoadingView() {
        if (isAttach()) {
            return view.get();
        } else {
            return null;
        }
    }
}

File: base/src/main/java/com/zhangteng/base/mvp/base/BaseLoadingPresenter.kt
package com.zhangteng.base.mvp.base

open class BaseLoadingPresenter<V : BaseLoadingView<*>, M : IModel> : BasePresenter<V, M>() {
    open fun getBaseLoadingView(): V? {
        return if (isAttach()) {
            mView?.get()
        } else {
            null
        }
    }
}